src/AgentVisibilityContext.tsx

```diff
@@ -0,0 +1,107 @@
+import React, { createContext } from 'react';
+import { isEqual } from 'underscore';
+import {
+  DisplayAction,
+  ActiveState,
+  VisibilitySelectionMap,
+} from './constants';
+import {
+  getPayloadForHideAll,
+  getSelectionStateInfoPayload,
+} from './selectors';
+import { SelectionStateInfo, UIDisplayData } from '@aics/simularium-viewer';
+
+interface VisibilityContextType {
+  uiDisplayData: UIDisplayData;
+  selectionStateInfo: SelectionStateInfo;
+  allAgentsHiddenState: ActiveState;
+  receiveUIDisplayData: (data: UIDisplayData) => void;
+  updateVisibilityAndSelection: (
+    newValue: VisibilitySelectionMap,
+    hideOrHighlight: DisplayAction
+  ) => void;
+  setAllAgentVisibility: (hiddenState: ActiveState) => void;
+}
+const { Inactive, Active, Indeterminate } = ActiveState;
+
+export const VisibilityContext = createContext<VisibilityContextType>({
+  uiDisplayData: [],
+  selectionStateInfo: {
+    hiddenAgents: [],
+    highlightedAgents: [],
+    colorChange: null,
+  },
+  allAgentsHiddenState: Inactive,
+  receiveUIDisplayData: () => {},
+  updateVisibilityAndSelection: () => {},
+  setAllAgentVisibility: () => {},
+});
+
+export const VisibilityProvider = ({
+  children,
+}: {
+  children: React.ReactNode;
+}) => {
+  const [uiDisplayData, setuiDisplayData] = React.useState<UIDisplayData>([]);
+  const [selectionStateInfo, updateSelectionStateInfo] =
+    React.useState<SelectionStateInfo>({
+      hiddenAgents: [],
+      highlightedAgents: [],
+      colorChange: null,
+    });
+  const [allAgentsHiddenState, setHiddenState] =
```

Comment:
`allAgentsHiddenState`. It sounds like it's going to be a boolean. but it's a status. It also seems like something that could be derived instead of storing in state separately? If I'm reading this correctly it's whether or not all the agents are hidden?